src/Bicep.Core.Samples/Files/InvalidModules_LF/main.diagnostics.bicep

```diff
@@ -41,27 +41,27 @@ module './main.bicep' = {
 }
 
 module modANoName './modulea.bicep' = {
-//@[38:42) [BCP035 (Error)] The specified object is missing the following required properties: "name", "params". |{\n\n}|
+//@[0:6) [BCP035 (Error)] The specified object is missing the following required properties: "name", "params". |module|
```

Comment:
I wonder if we should special-case the error message for modules, resources and other declarations? I worry that highlighting the identifier and calling it an `object` may confuse people. Thoughts?

Reply:
Yeah I agree.

Reply:
What do you want to call it? "Declaration" is an internal name that wouldn't mean anything to me next to "module" or "resource". "block"? "top-level block"?

Reply:
never mind. figured out that the declarationParent's keyword.text has the actual name (module or resource, etc)

Reply:
Sorry jor joining this thread late. The message works in English but will cause us trouble when we localize all the strings in other languages in the future. In general it's always a problem when you have sentences with a non-localizable string in the middle. The way we typically avoid is is by having the "type" either before or after the non localizable text enclosed in double quotes, so changing the error message to something like this would work:
```csharp
$"The specified \"{blockName}\" declaration is missing the following required properties: {ToQuotedString(properties)}.");
```

I think it's ok to call it a declaration because it's a declarative language.


---
In reply to: [517251091](https://github.com/Azure/bicep/pull/767#discussion_r517251091) [](ancestors = 517251091)

Reply:
Btw, I'm going to add a way to easily set a baseline (basically discover and run all the copy commands automatically), so it's not tedious.

Reply:
@sebastus When #841 gets merged, it should be much easier to get tests fixed 😊.